[Bug]: Error: Plugin did not respond / crashed

### What happened?

when running terraform apply the plugin crashed if the configuration contains the following error.

resource "hcloud_server" "xyzzy" {

  name       = var.hcloud_server_name
  image      = var.hcloud_server_image
  server_type = var.hcloud_server_type
  ssh_keys   = [ var.hcloud_ssh_key_name ] <---- **hcloud_ssh_key_name = ""**

  user_data  = data.template_file.cloud_init.rendered

}

please issue a helpful hint that the ssh-key configuration is incorrect.


### What did you expect to happen?


in my case this error occurs because i read the ssh-key name from an environment variable and this is empty in this case. the plugin can't handle this. 

terraform plan should also detect this error, but unfortunately accepts the incorrect configuration: 

Output of terraform plan
      + server_type                = "cx11"
      + ssh_keys                   = [
          + "",
        ]
      + status                     = (known after apply)

reported status Ok

### Please provide a minimal working example

--- Stacktrace:

Stack trace from the terraform-provider-hcloud_v1.42.0 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 56 [running]:
github.com/hetznercloud/terraform-provider-hcloud/internal/server.getSSHkeys({0x102e16c48, 0x140002814a0}, 0x140005fdb00, 0x140001eb635?)
	github.com/hetznercloud/terraform-provider-hcloud/internal/server/resource.go:985 +0x278
github.com/hetznercloud/terraform-provider-hcloud/internal/server.resourceServerCreate({0x102e16c48, 0x140002814a0}, 0x0?, {0x102d58820?, 0x140005fdb00?})
	github.com/hetznercloud/terraform-provider-hcloud/internal/server/resource.go:316 +0x4f4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x140005a2380, {0x102e16c80, 0x1400008d680}, 0xd?, {0x102d58820, 0x140005fdb00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:733 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x140005a2380, {0x102e16c80, 0x1400008d680}, 0x14000677d40, 0x14000420580, {0x102d58820, 0x140005fdb00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:864 +0x890
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000523950, {0x102e16c80?, 0x1400008d320?}, 0x140005eaf00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:1024 +0xb70
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000422140, {0x102e16c80?, 0x1400052d2c0?}, 0x14000201e30)
	github.com/hashicorp/terraform-plugin-go@v0.16.0/tfprotov5/tf5server/server.go:821 +0x3b8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x102dd7680?, 0x14000422140}, {0x102e16c80, 0x1400052d2c0}, 0x14000201dc0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.16.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:422 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000536000, {0x102e1b4c0, 0x140002964e0}, 0x14000304000, 0x1400052de00, 0x10337bc60, 0x0)
	google.golang.org/grpc@v1.56.0/server.go:1337 +0xc64
google.golang.org/grpc.(*Server).handleStream(0x14000536000, {0x102e1b4c0, 0x140002964e0}, 0x14000304000, 0x0)
	google.golang.org/grpc@v1.56.0/server.go:1714 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.56.0/server.go:959 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.56.0/server.go:957 +0x16c

Error: The terraform-provider-hcloud_v1.42.0 plugin crashed!

--- a/internal/e2etests/server/resource_test.go
+++ b/internal/e2etests/server/resource_test.go
@@ -916,6 +916,37 @@ func TestServerResource_Protection(t *testing.T) {
 	})
 }
 
+func TestServerResource_EmptySSHKey(t *testing.T) {
+	// Regression Test for https://github.com/hetznercloud/terraform-provider-hcloud/issues/727
+	var srv hcloud.Server
+
+	srvRes := &server.RData{
+		Name:    "server-empty-ssh-key",
+		Type:    e2etests.TestServerType,
+		Image:   e2etests.TestImage,
+		SSHKeys: []string{"\"\""},
+	}
+	srvRes.SetRName("server-empty-ssh-key")
+
+	tmplMan := testtemplate.Manager{}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          e2etests.PreCheck(t),
+		ProviderFactories: e2etests.ProviderFactories(),
+		CheckDestroy:      testsupport.CheckResourcesDestroyed(server.ResourceType, server.ByID(t, &srv)),
+		Steps: []resource.TestStep{
+			{
+				// Create a new Server using the required values
+				// only.
+				Config: tmplMan.Render(t,
+					"testdata/r/hcloud_server", srvRes,
+				),
+				ExpectError: regexp.MustCompile("Invalid ssh key passed"),
+			},
+		},
+	})
+}
+
 func isRecreated(new, old *hcloud.Server) func() error {
 	return func() error {
 		if new.ID == old.ID {

--- a/internal/server/resource.go
+++ b/internal/server/resource.go
@@ -92,7 +92,15 @@ func Resource() *schema.Resource {
 			"ssh_keys": {
 				Type:     schema.TypeList,
 				Optional: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
+				Elem: &schema.Schema{Type: schema.TypeString,
+					ValidateDiagFunc: func(i interface{}, _ cty.Path) diag.Diagnostics {
+						s := i.(string)
+						if len(s) == 0 {
+							return diag.Errorf("Invalid ssh key passed. You need to pass a string with at least 1 character")
+						}
+
+						return nil
+					}},
 				ForceNew: true,
 			},
 			"keep_disk": {

